apis/python/src/tiledbsoma/common_nd_array.py

```diff
@@ -97,10 +109,20 @@ def _build_tiledb_schema(
     dims = []
     for n, e in enumerate(shape):
         dim_name = f"soma_dim_{n}"
+
+        slot_capacity = 2**63 if e is None else e
```

Comment:
see comment above - it feels to me that this defaulting belongs in the sparse subclass

maybe worth asking @thetorpedodog .

Reply:
@bkmartinjr we had code that was de-duplicated and now it needs to be re-duplicated

i can move the code around, no problem

Reply:
…so that it looks something like

```python
# Here...
for n, size in enumerate(shape):
  # ...
  domain, extent = cls._dim_domain_extent(size)
  dims.append(tiledb.Dim(
    name=f"soma_dim_{n}",
    domain=(0, domain),
    extent=extent,
  ))

# Elsewhere...

class DenseNDArray:
  @classmethod
  def _dim_domain_extent(cls, name: str, size: Optional[int]) -> Tuple[int, int]:
    if not isinstance(size, int) or size <= 0:
      raise ValueError(...)
    return ...

class SparseNDArray:
  @classmethod
  def _dim_domain_extent(cls, str, size):
    ...
```

Reply:
done